components/BottomNavigation/examples/BottomNavigationTitleVisibilityChangeExample.swift

```diff
@@ -0,0 +1,117 @@
+// limitations under the License.
+
+import Foundation
+import MaterialComponents.MaterialBottomNavigation_ColorThemer
+import MaterialComponents.MaterialColorScheme
+
+class BottomNavigationTitleVisibilityChangeExample: UIViewController, MDCBottomNavigationBarDelegate {
+  
+  var colorScheme = MDCSemanticColorScheme()
+  
+  // Create a bottom navigation bar to add to a view.
+  let bottomNavBar = MDCBottomNavigationBar()
+  
+  init() {
+    super.init(nibName: nil, bundle: nil)
+    commonBottomNavigationTypicalUseSwiftExampleInit()
+  }
+  
+  @available(*, unavailable)
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+  
+  func commonBottomNavigationTypicalUseSwiftExampleInit() {
+    view.backgroundColor = .lightGray
+    view.addSubview(bottomNavBar)
+    
+    // Always show bottom navigation bar item titles.
+    bottomNavBar.titleVisibility = .always
+    
+    // Cluster and center the bottom navigation bar items.
+    bottomNavBar.alignment = .centered
+    
+    // Add items to the bottom navigation bar.
+    let tabBarItem1 = UITabBarItem(title: "Home", image: UIImage(named: "Home"), tag: 0)
+    let tabBarItem2 =
+      UITabBarItem(title: "Messages", image: UIImage(named: "Email"), tag: 1)
+    let tabBarItem3 =
+      UITabBarItem(title: "Favorites", image: UIImage(named: "Favorite"), tag: 2)
+    bottomNavBar.items = [ tabBarItem1, tabBarItem2, tabBarItem3 ]
+    
+    // Select a bottom navigation bar item.
+    bottomNavBar.selectedItem = tabBarItem2;
+    
+    bottomNavBar.delegate = self
+  }
+  
+  func layoutBottomNavBar() {
+    let size = bottomNavBar.sizeThatFits(view.bounds.size)
+    var bottomNavBarFrame = CGRect(x: 0,
+                                   y: view.bounds.height - size.height,
+                                   width: size.width,
+                                   height: size.height)
+    if #available(iOS 11.0, *) {
+      bottomNavBarFrame.size.height += view.safeAreaInsets.bottom
+      bottomNavBarFrame.origin.y -= view.safeAreaInsets.bottom
+    }
+    bottomNavBar.frame = bottomNavBarFrame
+  }
+  
+  func addInstructionLabel() {
+    let instructionLabel = UILabel()
+    instructionLabel.numberOfLines = 0
+    instructionLabel.lineBreakMode = .byWordWrapping
+    instructionLabel.text = "Choose the Home tab to make all titles disappear, and any other tab to make them reappear."
+    view.addSubview(instructionLabel)
+    let size = instructionLabel.sizeThatFits(view.bounds.size)
+    let instructionFrame = CGRect(x: 0,
+                                  y: view.bounds.height / 2 - size.height / 2,
+                                  width: size.width,
+                                  height: size.height)
+    instructionLabel.frame = instructionFrame
+  }
+  
+  override func viewWillLayoutSubviews() {
+    super.viewWillLayoutSubviews()
+    layoutBottomNavBar()
+    addInstructionLabel()
```

Comment:
I may be mistaken, but this looks like it creates a new UILabel every time the view is resized or rotated?   Did you mean to put the call to `addInstructionLabel()` in `viewDidLoad`?

Reply:
Thank you for spotting that. Moved to viewDidLoad().